### PR TITLE
Revising Taxi fare regression tutorial sample for ML.NET .7 release

### PR DESCRIPTION
**Pull request by JRAlexander**
Revising the Taxi fare regression tutorial sample for ML.NET .7 release. 

Tutorial is https://github.com/dotnet/docs/pull/8863

### CODE REVIEW

**Comment by BillWagner** [comment]
@JRAlexander 

I read through all the code, and overall it looks good. Do you know why none of the File I/O operations are async? Is that the intended design?

**Comment by JRAlexander** [comment]
@billwagner, I'm not sure, may be just an oversight. @CESARDELATORRE, @GalOshri or @shmoradims - any thoughts?

File: machine-learning/tutorials/TaxiFarePrediction/Program.cs [hunk 1]
```diff
@@ -1,97 +1,179 @@
...
         {
+            // <Snippet6>
+            IDataView dataView = _textLoader.Read(new MultiFileSource(dataPath));
+            // </Snippet6>
+
+            // <Snippet7>
+            var pipeline = new CopyColumnsEstimator(mlContext, "FareAmount", "Label")
+            // </Snippet7>
+                    // <Snippet8>
+                    .Append(new OneHotEncodingEstimator(mlContext, "VendorId"))
```
Comment (shmoradims) [inline machine-learning/tutorials/TaxiFarePrediction/Program.cs]: >new OneHotEncodingEstimator(mlContext [](start = 28, length = 37)

mlContext.Transforms.Categorical.OneHotEncoding(

File: machine-learning/tutorials/TaxiFarePrediction/Program.cs [hunk 2]
```diff
@@ -1,97 +1,179 @@
...
+                    .Append(new OneHotEncodingEstimator(mlContext, "RateCode"))
+                    .Append(new OneHotEncodingEstimator(mlContext, "PaymentType"))
+                    // </Snippet8>
+                    // <Snippet9>
+                    .Append(new ColumnConcatenatingEstimator(mlContext, "Features", "VendorId", "RateCode", "PassengerCount", "TripTime", "TripDistance", "PaymentType"))
```
Comment (shmoradims) [inline machine-learning/tutorials/TaxiFarePrediction/Program.cs]: >new ColumnConcatenatingEstimator [](start = 28, length = 32)

mlContext.Transforms.Concatenate

File: machine-learning/tutorials/TaxiFarePrediction/Program.cs [hunk 3]
```diff
@@ -1,97 +1,179 @@
...
+                    // </Snippet9>
+                    // <Snippet10>
+                    .Append(mlContext.Regression.Trainers.FastTree("Label", "Features"));
```
Comment (shmoradims) [inline machine-learning/tutorials/TaxiFarePrediction/Program.cs]: >"Label", "Features" [](start = 67, length = 19)

defaults. keep or remove?

File: machine-learning/tutorials/TaxiFarePrediction/Program.cs [hunk 4]
```diff
@@ -1,97 +1,179 @@
...
-            // <Snippet13>
-            var evaluator = new RegressionEvaluator();
-            RegressionMetrics metrics = evaluator.Evaluate(model, testData);
-            // </Snippet13>
-
+        private static void Evaluate(MLContext mlContext, ITransformer model)
+        {
             // <Snippet14>
-            Console.WriteLine($"Rms = {metrics.Rms}");
+            IDataView dataView = _textLoader.Read(new MultiFileSource(_testDataPath));
```
Comment (shmoradims) [inline machine-learning/tutorials/TaxiFarePrediction/Program.cs]: >new MultiFileSource(_testDataPath) [](start = 50, length = 34)

Read(_testDataPath)

File: machine-learning/tutorials/TaxiFarePrediction/Program.cs [hunk 5]
```diff
@@ -1,97 +1,179 @@
...
+            using (var stream = new FileStream(_modelPath, FileMode.Open, FileAccess.Read, FileShare.Read))
+            {
+                loadedModel = TransformerChain.LoadFrom(mlContext, stream);
+            }
+            // </Snippet20>
+
+            //Prediction test
+            // Create prediction engine and make prediction.
+            // <Snippet21>
+            var engine = loadedModel.MakePredictionFunction<TaxiTrip, TaxiTripFarePrediction>(mlContext);
```
Comment (shmoradims) [inline machine-learning/tutorials/TaxiFarePrediction/Program.cs]: >engine [](start = 16, length = 6)

let's call it predictionFunction

File: machine-learning/tutorials/TaxiFarePrediction/Program.cs [hunk 6]
```diff
@@ -1,97 +1,179 @@
...
+            Console.WriteLine($"Predicted fare: {prediction.FareAmount:0.####}, actual fare: 15.5");
+            Console.WriteLine($"**********************************************************************");
+            // </Snippet24>
+        }
+
+        private static void SaveModelAsFile(MLContext mlContext, ITransformer model)
+        {
+            // <Snippet13> 
+            using (var fs = new FileStream(_modelPath, FileMode.Create, FileAccess.Write, FileShare.Write))
+                model.SaveTo(mlContext, fs);
```
Comment (shmoradims) [inline machine-learning/tutorials/TaxiFarePrediction/Program.cs]: >model.SaveTo(mlConte [](start = 16, length = 20)

use mlContext

**Pull request merged**